File: 0019-remove-nth-node-from-end-of-list/0019-remove-nth-node-from-end-of-list.ts
/**
 * Definition for singly-linked list.
 * class ListNode {
 *     val: number
 *     next: ListNode | null
 *     constructor(val?: number, next?: ListNode | null) {
 *         this.val = (val===undefined ? 0 : val)
 *         this.next = (next===undefined ? null : next)
 *     }
 * }
 */

function removeNthFromEnd(head: ListNode | null, n: number): ListNode | null {
    let dummy = new ListNode(0, head);
    let goUptoN = head;
    let JumpN = dummy;
    for (let i = 0; i < n; i++) {
        goUptoN = goUptoN.next;
    }
    while (goUptoN != null) {
        goUptoN = goUptoN.next;
        JumpN = JumpN.next;
    }
    JumpN.next = JumpN.next.next;
    return dummy.next;
};

File: 0019-remove-nth-node-from-end-of-list/0019-remove-nth-node-from-end-of-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def removeNthFromEnd(self, head: Optional[ListNode], n: int) -> Optional[ListNode]:
        d=head
        lis=[]
        while d:
            if not d: return
            lis.append(d)
            d=d.next
         
        held=[]
        
        N = len(lis)
        skip_index = N - n
 
        if skip_index == 0:
            return head.next 
        lis[skip_index - 1].next = lis[skip_index].next if skip_index + 1 < N else None

        return head
        
            
        